### ISSUE TITLE
useStoryblokState returns non-undefined for undefined parameter

### ISSUE THREAD

**Issue by fabb** [storyblok/storyblok-react#259]
We might have a next.js page like this:

```
export const getServerSideProps = async (context) => {
    const pageArticleStory = await getStory('1234567890')
        .catch((error) => {
            console.log(error)
            // graceful fallback in case cms is not available
            return undefined
        })

    return {
        props: {
            pageArticleStory: pageArticleStory?.story ?? null,
        },
    }
}

export default const Page = (props) => {
    return (
        <Fragment>
            {props.pageArticleStory ? (
                <Article story={props.pageArticleStory} />
            ) : (
                <div>Fallback if story is not available</div>
            )}
        </Fragment>
    )
}
```

This code works fine both when the story can be fetched from cms and when fetching failed. But when adding support for the live preview mode with `useStoryblokState`, we have a problem:

```
export default const Page = (props) => {
    const pageArticleStory = useStoryblokState(props.pageArticleStory ?? undefined)

    return ...
```

The problem is when `props.pageArticleStory` is `null`, then the return value of `useStoryblokState` is still not `undefined`, but it is a story with an `undefined` `content`.
```
    useStoryblokState(undefined) !== undefined
    useStoryblokState(undefined)?.content === undefined
```

This is a problem because the ternary for rendering a fallback no longer works (`props.pageArticleStory ? ... : <Fallback />`) and tries to render the `Article` component which then crashes at runtime.

So in this issue I request to change `useStoryblokState` so it returns `undefined` when its parameter is `undefined` to fix this behavior:
```
    useStoryblokState(undefined) === undefined
```

**Comment by fgiuliani** [comment]
Hello @fabb !
In the code of `useStoryblokState` function, you can see the function returns null when the story is null.

https://github.com/storyblok/storyblok-react/blob/main/lib/index.ts#L76

You could do something like this:

https://github.com/storyblok/getting-started/blob/master/reactjs/src/App.js#L12

Does this work for you? Or probably I'm not following the issue 🤔

**Comment by fgiuliani** [comment]
Hey @fabb I created a PR:
https://github.com/storyblok/storyblok-react/pull/319

Where I add a change that, I think, should help with your request:
https://github.com/storyblok/storyblok-react/pull/319/files#diff-d5a4b3a0309f2337144861084c1014523cced561eb722cb9684c63f5e41e0567R63

What do you think?

**Comment by fgiuliani** [comment]
@fabb changed applied!

**Comment by fgiuliani** [comment]
Hello again!
We just released Storyblok React SDK v2.0.0:
https://github.com/storyblok/storyblok-react/releases/tag/v2.0.0

In that version, we are adding the previously mentioned change. So, this issue should be fixed now. 
Let us know if this works for you.

**Comment by fgiuliani** [comment]
Hey @fabb thanks for pointing that and creating the PR. I left a comment on it
